public/app/core/components/PageNew/Page.tsx

```diff
@@ -30,19 +29,20 @@ export const Page: PageType = ({
 }) => {
   const styles = useStyles2(getStyles);
   const navModel = usePageNav(navId, oldNavProp);
+  const { chrome } = useGrafanaInternal();
 
   usePageTitle(navModel, pageNav);
 
   const pageHeaderNav = pageNav ?? navModel?.node;
 
   useEffect(() => {
     if (navModel) {
-      appChromeService.update({
+      chrome.update({
         sectionNav: navModel.node,
         ...(pageNav && { pageNav }),
       });
     }
-  }, [navModel, pageNav]);
+  }, [navModel, pageNav, chrome]);
```

Comment:
@dprokop it can't be stored locally, it's used from other components (GrafanaRoute and AppChromeUpdate, Page)

Reply:
@torkelo I just find it redundant to include service dependencies in the `useEffect` deps list, if we know that those don't change over time. It may be me, but when looking at `useEffect` I usually inspect it's deps to understand when it's gone evaluate. In this case, the service dep is noise. 

What about the idea I posted in a comment below, but instead returning `ref.current` you would just return refs from the hook? Sth like this: https://stackblitz.com/edit/react-ts-9ketyn?file=index.tsx

I know that it brings the need to use `ref.current`, but this is neatly handled by TS identifying that the returned service is a reference.

Reply:
I know it's a bit old thread but as I am working on this I am curious, @dprokop why the need to remove the need of adding the services to dependency list? Imho the reason for not making the services singletons is that you want to have different services when needed (like for the sidecar project) making the return type a ref would imho create possible errors because in that case the service can actually change and components should react to that.